pkg/controller/infrastructure/infraflow/access/networking_access.go

```diff
@@ -141,7 +143,7 @@ func (a *networkingAccess) GetRouterByID(id string) (*Router, error) {
 		return nil, err
 	}
 	if len(routers) == 0 {
-		return nil, nil
+		return nil, fmt.Errorf("missing expected router %s", id)
```

Comment:
Doesn't this change the codepath for the following?
https://github.com/gardener/gardener-extension-provider-openstack/blob/c14529a1d7c69e071a077ae1c88d67a24a75eab9/pkg/controller/infrastructure/infraflow/reconcile.go#L180

Reply:
Yes, thanks for the comment